pkg/tracking/store/sql/tags.go

```diff
@@ -80,3 +81,182 @@ func (s TrackingSQLStore) setTagsWithTransaction(
 
 	return nil
 }
+
+const (
+	maxEntityKeyLength = 250
+	maxTagValueLength  = 8000
+)
+
+// Helper function to validate the tag key and value
+func validateTag(key, value string) *contract.Error {
+	if key == "" {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			"Missing value for required parameter 'key'",
+		)
+	}
+	if len(key) > maxEntityKeyLength {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf("Tag key '%s' had length %d, which exceeded length limit of %d", key, len(key), maxEntityKeyLength),
+		)
+	}
+	if len(value) > maxTagValueLength {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf("Tag value exceeded length limit of %d characters", maxTagValueLength),
+		)
+	}
+	// TODO: Check if this is the correct way to prevent invalid values
+	if _, err := strconv.ParseFloat(value, 64); err == nil {
```

Comment:
Where is this coming from? A float as tag value is weird, but I would not say it is invalid perse.

Reply:
This is also due to my misunderstanding in how validations happen! The comment below shows my reasoning for adding the validation.